src/main/java/Duke.java

```diff
@@ -1,10 +1,134 @@
+                listTasks();
+            } else if(task.contains("mark ")&& !task.contains("unmark ")) {
+                markTask(task);
+            } else if(task.contains("unmark ")) {
+                unmarkTask(task);
+            } else {
+                addTask(task);
+            }
+            System.out.println("    ____________________________________");
+        }
+    }
+
+    private static void greetUser(String greeting) {
+        if(greeting.equals("hi")) {
+            String logo = " ____        _\n"
+                    + "|  _ \\ _   _| | _____\n"
+                    + "| | | | | | | |/ / _ \\\n"
+                    + "| |_| | |_| |   <  __/\n"
+                    + "|____/ \\__,_|_|\\_\\___|\n";
+            System.out.println(logo);
+            System.out.println("    ____________________________________");
+            System.out.println("    Hello! I'm Duke");
+            System.out.println("    What can I do for you?\n");
+        }
+        else {
+            System.out.println("    ____________________________________");
+            System.out.println("    Bye. Hope to see you again soon!");
+            System.out.println("    ____________________________________");
+        }
+    }
+
+    public static void listTasks() {
+        System.out.println("    Here are the tasks in your list:");
+        for(int i=0;i< taskList.length;i++) {
+            if(taskList[i] == null) break;
+            System.out.println("    "+(i+1)+": "+taskList[i]);
+        }
+    }
+
+    public static void markTask(String task) {
+        task = task.replace("mark ","");
+        int i = Integer.parseInt(task)-1;
+        if(taskList[i]!=null) {
+            taskList[i].setDone(true);
+            System.out.println("    Nice! I've marked this task as done:");
+            System.out.println("    "+taskList[i]);
+        } else {
+            System.out.println("    Please enter a valid task number");
+        }
+    }
+
+    public static void unmarkTask(String task) {
+        task = task.replace("unmark ","");
+        int i = Integer.parseInt(task)-1;
+        if(taskList[i]!=null) {
+            taskList[i].setDone(false);
+            System.out.println("    OK, I've marked this task as not done yet:");
+            System.out.println("        "+taskList[i]);
+
+        } else {
+            System.out.println("    Please enter a valid task number");
+        }
+    }
+
+    public static void addTask(String task) {
+        int index = task.indexOf(' ');
+        String taskType = task.substring(0,index);
+        task = task.substring(index+1);
+        switch(taskType) {
+        case "todo":
+            addAsTodo(task);
+            break;
+        case "deadline":
+            addAsDeadline(task);
+            break;
+        case "event":
+            addAsEvent(task);
+            break;
+        default:
+            System.out.println("    Sorry I do not know what that means");
+            return;
```

Comment:
It is pretty wise to split three different tasks into separate functions, it makes the code much neater. The way you catch the error is good also.